async-race/src/store/garageStore.ts

```diff
@@ -0,0 +1,149 @@
+import { defineStore } from "pinia";
+import type { Car, DriveOptions } from "../common/model";
+import { getCarBrand, getRandomColor } from "../common/functions";
+import { ref, watch } from "vue";
+import {
+  createCar,
+  deleteCar,
+  getCar,
+  getGarage,
+  startEngine,
+  stopEngine,
+  switchEngine,
+  updateCar,
+} from "../utils/engine";
+
+export const useGarageStore = defineStore("garage", () => {
+  const garage = ref<Car[]>([]);
+  const limit = 7;
+  const page = ref(1);
+  const selectedCarId = ref<number | null>(null);
+  const driveOptions = ref<DriveOptions[]>([]);
+  const carDriveStatus = ref<true>();
+  const car = ref<Car>();
+
+  watch(page, () => {
+    getCars();
+  });
+
+  function nextPage() {
+    page.value += 1;
+  }
+
+  function previousPage() {
+    page.value -= 1;
+  }
+
+  function selectCar(id: number) {
+    selectedCarId.value = id;
+  }
+
+  function startAllEngines() {
+    garage.value.forEach((car) => startCarEngine(car.id));
+  }
+
+  function stopAllEngines() {
+    garage.value.forEach((car) => stopCarEngine(car.id));
+  }
+
+  function startCarEngine(id: number) {
+    driveOptions.value = [];
+    carDriveStatus.value = undefined;
+
+    startEngine(id).then((value) => {
+      const time = value.distance / value.velocity;
+
+      driveOptions.value = [
+        ...driveOptions.value,
+        {
+          time,
+          startedStatus: true,
+          driveStatus: false,
+          resetStatus: false,
+          id,
+        },
+      ];
+    });
+
+    switchEngine(id)
+      .catch(() => {
+        const findedCar = driveOptions.value.findIndex((car) => car.id === id);
+        driveOptions.value[findedCar].startedStatus = false;
+      })
+      .then((value) => {
+        const findedCar = driveOptions.value.findIndex((car) => car.id === id);
+
+        if (value) {
+          driveOptions.value[findedCar].driveStatus = true;
+        }
+      });
+  }
+
+  function stopCarEngine(id: number) {
+    stopEngine(id).then(() => {
+      const findedCar = driveOptions.value.findIndex((car) => car.id === id);
+      driveOptions.value[findedCar].resetStatus = true;
+    });
+  }
+
+  function getCars() {
+    getGarage(limit, page.value).then((cars) => {
+      garage.value = cars;
+    });
+    return garage;
+  }
+
+  function getOneCar(id: number) {
+    getCar(id).then((value) => {
+      car.value = value;
+    });
+    return car.value;
+  }
+
+  function generateCar(name: string, color: string) {
+    createCar(name, color);
+    getCars();
+  }
+
+  function createCars() {
+    for (let i = 0; i < 100; i++) {
+      generateCar(getCarBrand(), getRandomColor());
+    }
+  }
+
+  function removeCar(id: number) {
+    deleteCar(id);
+    getCars();
+  }
+
+  function updatCar(name: string, color: string) {
+    if (selectedCarId.value === null) {
+      return;
+    }
+
+    updateCar(name, color, selectedCarId.value);
+    getCars();
```

Comment:
```suggestion
updateCar(name, color, selectedCarId.value).then(getCars)
```

тоже самое что и выше